sklearn/model_selection/tests/test_split.py

```diff
@@ -0,0 +1,717 @@
+    assert_raises(ValueError, ShuffleSplit, test_size=2.0)
+    assert_raises(ValueError, ShuffleSplit, test_size=1.0)
+    assert_raises(ValueError, ShuffleSplit, test_size=0.1, train_size=0.95)
+    assert_raises(ValueError, next, ShuffleSplit(test_size=11).split(X))
+    assert_raises(ValueError, next, ShuffleSplit(test_size=10).split(X))
+    assert_raises(ValueError, next, ShuffleSplit(test_size=8,
+                                                 train_size=3).split(X))
+    assert_raises(ValueError, ShuffleSplit, test_size=None, train_size=None)
+    assert_raises(ValueError, next, ShuffleSplit(train_size=1j).split(X))
+
+
+def test_shufflesplit_reproducible():
+    # Check that iterating twice on the ShuffleSplit gives the same
+    # sequence of train-test when the random_state is given
+    ss = ShuffleSplit(random_state=21)
+    assert_array_equal(list(a for a, b in ss.split(X)),
+                       list(a for a, b in ss.split(X)))
+
+
+def test_safe_split_with_precomputed_kernel():
+    clf = SVC()
+    clfp = SVC(kernel="precomputed")
+
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    K = np.dot(X, X.T)
+
+    cv = ShuffleSplit(test_size=0.25, random_state=0)
+    tr, te = list(cv.split(X))[0]
+
+    X_tr, y_tr = _safe_split(clf, X, y, tr)
+    K_tr, y_tr2 = _safe_split(clfp, K, y, tr)
+    assert_array_almost_equal(K_tr, np.dot(X_tr, X_tr.T))
+
+    X_te, y_te = _safe_split(clf, X, y, te, tr)
+    K_te, y_te2 = _safe_split(clfp, K, y, te, tr)
+    assert_array_almost_equal(K_te, np.dot(X_te, X_tr.T))
+
+
+def test_train_test_split_allow_nans():
+    # Check that train_test_split allows input data with NaNs
+    X = np.arange(200, dtype=np.float64).reshape(10, -1)
+    X[2, :] = np.nan
+    y = np.repeat([0, 1], X.shape[0] / 2)
+    train_test_split(X, y, test_size=0.2, random_state=42)
+
+
+def test_check_cv_return_types():
+    X = np.ones((9, 2))
+    cv = check_cv(3, classifier=False)
+    # Use numpy.testing.assert_equal which recursively compares
+    # lists of lists
+    np.testing.assert_equal(list(KFold(3).split(X)), list(cv.split(X)))
+
+    y_binary = np.array([0, 1, 0, 1, 0, 0, 1, 1, 1])
+    cv = check_cv(3, y_binary, classifier=True)
+    np.testing.assert_equal(list(StratifiedKFold(3).split(X, y_binary)),
+                            list(cv.split(X, y_binary)))
+
+    y_multiclass = np.array([0, 1, 0, 1, 2, 1, 2, 0, 2])
+    cv = check_cv(3, y_multiclass, classifier=True)
+    np.testing.assert_equal(list(StratifiedKFold(3).split(X, y_multiclass)),
+                            list(cv.split(X, y_multiclass)))
+
+    X = np.ones((5, 2))
+    y_seq_of_seqs = [[], [1, 2], [3], [0, 1, 3], [2]]
+
+    with warnings.catch_warnings(record=True):
+        # deprecated sequence of sequence format
+        cv = check_cv(3, y_seq_of_seqs, classifier=True)
+    np.testing.assert_equal(list(KFold(3).split(X)), list(cv.split(X)))
+
+    y_indicator_matrix = MultiLabelBinarizer().fit_transform(y_seq_of_seqs)
```

Comment:
yes it does the same thing! seq of seq (multilabel) input to `LabelBinarizer` was deprecated and so I used `MultiLabelBinarizer`...

Thanks for pointing this out... seq of seq support for LabelBinarizer is earmarked for removal in 0.17... should we remove it??